cmd/cli/serve/serve.go

```diff
@@ -123,12 +124,29 @@ func serve(cmd *cobra.Command, cfg types.Bacalhau, fsRepo *repo.FsRepo) error {
 		return err
 	}
 
+	installationID, err := fsRepo.ReadInstallationID()
+	if err != nil {
+		return err
+	}
+
 	computeConfig, err := GetComputeConfig(ctx, cfg, isComputeNode)
 	if err != nil {
 		return errors.Wrapf(err, "failed to configure compute node")
 	}
 
-	requesterConfig, err := GetRequesterConfig(cfg, isRequesterNode)
+	var recorder analytics.Recorder
+	if cfg.DisableAnalytics {
+		recorder = &analytics.NoopRecorder{}
+	} else {
+		recorder, err = analytics.New(ctx, analytics.WithNodeNodeID(nodeName),
+			analytics.WithInstallationID(installationID), analytics.WithNodeType(isRequesterNode, isComputeNode))
+		if err != nil {
+			// if we fail here for some reason, do so silently and use the noop recorder.
+			recorder = &analytics.NoopRecorder{}
+		}
+	}
+
+	requesterConfig, err := GetRequesterConfig(cfg, isRequesterNode, recorder)
```

Comment:
I didn't use a global here since we'd like to only create a recorder if the configuration allows it.
Our logger, and other otel providers can all be configured with environment variables which makes it easier to configure them inside an `init()` method. 
I'll leave the decision here up to you.

Reply:
I know you didn't use globals in your implementation, but it is a common practice and we are already doing it today with setting otel trace and metric providers

https://github.com/bacalhau-project/bacalhau/blob/246a2bdc0dce46233a23727386f55c9f602c929c/pkg/telemetry/metrics.go#L39

globals doesn't mean you set them inside an `init()`, but wherver you set them you then assign them to a global reference

Reply:
okay I will set it as the global provider.